docs: add examples how to run ig in a custom pod (#1887)
inspektor-gadget/inspektor-gadget: docs/ig.md

  `--auto-mount-filesystems` to ig to automatically mount those filesystems.
- `--pid=host` runs in the host PID namespace. Optional on Linux. This is necessary on Docker Desktop on Windows because
  /host/proc does not give access to the host processes.

### Using ig in a custom Kubernetes pod

> If you want to include `ig` in your own container images, you take example on the following Dockerfile:

Reviewer: you take -> you can take ?

```Dockerfile
# In production, you should use a specific version of ig instead of latest:
# --build-arg BASE_IMAGE=ghcr.io/inspektor-gadget/ig:v0.18.1
ARG BASE_IMAGE=ghcr.io/inspektor-gadget/ig:latest
FROM ${BASE_IMAGE} as ig

> # Your own image

Reviewer: What about mentioning that it should work fine in most cases but that the right libc version should be present for it to work?

Reviewer: Ah right, I usually use `go build .` on my machine and it doesn't produce a static binary, but `make ig` does. Sorry for the noise.

FROM alpine:3.17
COPY --from=ig /usr/bin/ig /usr/bin/ig
ENV HOST_ROOT=/host
# The rest of your Dockerfile

> ```

Reviewer 2: Isn't this also valid outside k8s?

> If you want to run `ig` in a Kubernetes pod, you can take example on

Reviewer: ```suggestion

In order to `ig` in a Kubernetes pod use 
```

  `--auto-mount-filesystems` to ig to automatically mount those filesystems.
- `--pid=host` runs in the host PID namespace. Optional on Linux. This is necessary on Docker Desktop on Windows because
  /host/proc does not give access to the host processes.

> ### Using ig in a Kubernetes pod

Reviewer: I'm wondering if we can add more information on this case. We should describe what is the motivation to run `ig` and not `ig-k8s`. (I suppose it's because by using ig we can get the output directly without needing to execute kubectl-gadget, right?)

In order to run `ig` in a Kubernetes pod use [examples/pod-ig.yaml](examples/pod-ig.yaml).

```bash
$ kubectl apply -f docs/examples/pod-ig.yaml
$ kubectl logs ig
RUNTIME.CONTAINERNAME          RUNTIME.CONTAIN… PID              PPID             COMM             RET ARGS
kube-proxy                     k8s.gcr.io/kube… 3985376          3024961          ip6tables        0   /usr/sbin/ip6tables -w 5 -W 100000 -S K…
```

### Adding ig in your own container image

> In order to add `ig` in your own container image, you can take example on the following Dockerfile:

Author: I added an explanation about multi-platform builds.
